src/System.Private.Uri/src/System/IPv6AddressHelper.cs

```diff
@@ -31,7 +31,8 @@ internal static string ParseCanonicalName(string str, int start, ref bool isLoop
                 ((long*)numbers)[0] = 0L;
                 ((long*)numbers)[1] = 0L;
                 isLoopback = Parse(str, numbers, start, ref scopeId);
-                return '[' + CreateCanonicalName(numbers) + ']';
+                linkLocalAddress = numbers[0] == 0xfe80;
+                return '[' + CreateCanonicalName(numbers) + (linkLocalAddress ? scopeId : "") + ']';
```

Comment:
Nit: the compiler will help fix this up, but it'd be better to use `"["` and `"["` rather than `'['` and `']'`.

Reply:
understood TYVM!